### PR TITLE
add basic read stream

### PR DESCRIPTION
**Pull request by mafintosh**
Really basic read stream so we have it. Will need the prefetching logic etc later on

### CODE REVIEW

File: lib/streams.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  async _readP () {
+    const end = this.live ? -1 : (this.end === -1 ? this.core.length : this.end)
+    if (end >= 0 && this.start >= end) {
+      this.push(null)
+      return
+    }
+
+    this.push(await this.core.get(this.start++))
+  }
```
Comment (kasperisager) [inline lib/streams.js:36]: Might it be worthwhile to add promise support to streamx at some point to avoid this kind of pattern?

Comment (mafintosh) [reply lib/streams.js:36]: yuh! actually made a PR, that I just need to finish

File: README.md [hunk 2]
```diff
@@ -98,6 +98,18 @@ Truncate the core to a smaller length.
...
+#### `const stream = core.createReadStream([options])`
+
+Make a read stream. Options include:
+
+``` js
+{
+  start: 0,
+  end: core.length,
+  live: false
+}
```
Comment (kasperisager) [inline README.md:110]: Is `opts.snapshot` public API that should be documented?

Comment (mafintosh) [reply README.md:110]: Ya, I'm just not ready to explain it at this stage 🤣

Comment (kasperisager) [reply README.md:110]: Fair enough 😂

Comment (mafintosh) [reply README.md]: I documented it now, it was a bad excuse :)

**Review by kasperisager — APPROVED** [review]
LGTM 👍

**Comment by mafintosh** [comment]
10.0.0-alpha.11

**Pull request merged**